Add an explanation of how plugins work to the manual (#5627)
ocaml/opam: doc/pages/Manual.md

- `<switch-prefix>/.opam-switch/backup`: snapshots of previous states of the
  switch, and other backup files.

#### Plugins

> When opam doesn't know a subcommand name (e.g. `opam plugin-name`), it looks at

Reviewer: ```suggestion
When opam doesn't know a subcommand name (e.g., `opam plugin-name`), it looks at
```

the binaries installed in `${OPAMROOT}/plugins/bin` (by default: `~/.opam/plugins/bin`).

> This directory contains symlinks to binaries installed in whichever opam switch
> the plugin it belongs to was installed in.

Reviewer: ```suggestion
This directory contains symlinks to binaries installed in the same directory where the plugin's opam switch
was installed.
```
Here is a suggestion that might be a little more clear (although I'm not sure if the plugin belongs to the opam switch or the opam switch belongs to the plugin). Clarify that, and we can reword as needed.

> This directory contains symlinks to binaries installed in whichever opam switch
> the plugin it belongs to was installed in.

Reviewer: ```suggestion
This directory contains symlinks to binaries installed in the same directory where the plugin's opam switch
was installed.
```
Here is a suggestion that might be a little more clear (although I'm not sure if the plugin belongs to the opam switch or the opam switch belongs to the plugin). Clarify that, and we can reword as needed.

These symlinks are added for any packages that have the ["plugin" flag](#opamflag-plugin)

> and have a binary matching their package name. If no binaries with the same name

Reviewer 2: I think the binary only has to match the opam package name by convention; it's not a hard requirement.

Author: it is a "hard" requirement. A warning will be displayed otherwise and the link will not appear.

> as the package is found, a warning will be shown.

Reviewer: ```suggestion
as the package are found, a warning will be shown.
```

> and have a binary matching their package name. If no binaries with the same name
> as the package is found, a warning will be shown.

Reviewer: ```suggestion
and have a binary matching their package name. If no binaries with a matching name are found, a warning will be shown.
```
